src/greeter/GreeterApp.cpp

```diff
@@ -247,11 +247,16 @@ int main(int argc, char **argv) {
     qInstallMessageHandler(SDDM::GreeterMessageHandler);
 
     // HiDPI
-    if (SDDM::mainConfig.EnableHiDPI.get()) {
+    bool hiDpiEnabled = false;
+    if (QGuiApplication::platformName() == QLatin1String("xcb"))
```

Comment:
an alternative is to make assumptions based on env vars but this way we are sure to check the correct platform